src/protocol/sort/compose.rs

```diff
@@ -44,24 +46,24 @@ impl<'a, F: Field> Compose<'a, F> {
     pub async fn execute(
         &mut self,
         ctx: ProtocolContext<'_, Replicated<F>, F>,
-    ) -> Result<(), BoxError> {
+    ) -> Result<Vec<Replicated<F>>, BoxError> {
         let mut random_permutations =
             get_two_of_three_random_permutations(self.rho.len(), &ctx.prss());
         let mut random_permutations_copy = random_permutations.clone();
 
-        Shuffle::new(self.sigma)
+        self.sigma = Shuffle::new(take(&mut self.sigma))
```

Comment:
This is cool. I learned about `std::mem::take()` today!

But if I understand it correctly, this will actually make a deep copy of the vector, and fully zero out the old one. That's a lot of wasted work. 

I think there is a better way:

Let's just make this protocol into a single function invocation. So instead of:
```
Compose::new(sigma, rho).execute(ctx).await?
```

Let's just do:
```
use crate::protocol::sort::compose::compose_permutations;

compose_permutations(ctx, sigma, rho).await?
```

This is the way I did it with `DoubleRandom::execute()` and `check_zero` and `reveal`. I just don't think creating the struct, assigning member variables, and referencing them from the `execute` method is adding anything. If anything, it's creating this added complexity for us where we have to call `std::mem::take()`.

Reply:
Actually mem::take or mem::swap dont do a deep copy but only copy the contents of the variable (i.e. copy location of vector data and length and, do not copy the actual data stored on the heap): https://stackoverflow.com/questions/39327859/how-to-swap-variables-pointing-to-data-without-swapping-their-content
I have also tried to consume the permutation variable in apply and apply_inv